draft-ietf-tls-esni.md

```diff
@@ -804,6 +804,17 @@ by attackers on the local network, which is a common case where ClientHello and
 encryption are desired. Moreover, as noted in the introduction, SNI encryption is
 less useful without encryption of DNS queries in transit via DoH or DPRIVE mechanisms.
 
+## Client Tracking
+
+A malicious client-facing server could distribute unique, per-client ECHOConfig
+structures as a way of tracking clients across subsequent connections. On-path
+adversaries which know about these unique keys could also track clients in this
+way by observing TLS connection attempts. This type of attack scales linearly
```

Comment:
A paragraph break here might be good. You jump from description of the attack to a description of mitigating factors without segue.